app/src/main/res/layout-land/popular_movies_fragment.xml

```diff
@@ -29,22 +29,32 @@
             app:layout_constraintStart_toStartOf="parent"
             app:layout_constraintTop_toTopOf="parent" />
 
-        <androidx.recyclerview.widget.RecyclerView
-            android:id="@+id/recyclerViewPopularMoviesFragment"
+        <androidx.swiperefreshlayout.widget.SwipeRefreshLayout
+            android:id="@+id/popularMoviesRefreshOnSwipe"
             android:layout_width="match_parent"
-            android:layout_height="match_parent"
-            android:adapter="@{recyclerAdapterForPopularMovies}"
-            android:clipToPadding="false"
-            android:paddingBottom="88dp"
-            app:adapterData="@{viewModel.popularMovies}"
-            app:layoutManager="androidx.recyclerview.widget.GridLayoutManager"
-            app:layout_constraintBottom_toBottomOf="parent"
-            app:layout_constraintEnd_toEndOf="parent"
-            app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toTopOf="parent"
-            app:spanCount="2"
-            tools:listitem="@layout/popular_movie_recycler_item_layout"
-            tools:visibility="visible" />
+            android:layout_height="wrap_content"
+            app:refreshing="@{viewModel.isStartingRefresh}"
+            app:onRefreshListener="@{() -> viewModel.refreshPopularMovie()}"
+            tools:ignore="MissingConstraints">
+
+            <androidx.recyclerview.widget.RecyclerView
+                android:id="@+id/recyclerViewPopularMoviesFragment"
+                android:layout_width="match_parent"
+                android:layout_height="match_parent"
+                android:adapter="@{recyclerAdapterForPopularMovies}"
```

Comment:
Just set it in Fragment, that would be shorter 😄

Reply:
Do you think about something like this in fragment:
```
binding.recyclerViewPopularMoviesFragment?.adapter = PopularMoviesRecyclerAdapter(popularMoviesFragmentViewModel)
```
and delete variable from XML 
```
<variable
            name="recyclerAdapterForPopularMovies"
            type="com.antosik.benchproject.app.popular.movies.view.PopularMoviesRecyclerAdapter" />
```
Could it make exceptions? I will set the adapter in fragment and data pass in XML I have an adapter for the recycler view for this.

Reply:
Ok, now I see what is problem. For one view I have a different id, for the recycler view for the vertical view I had `popularMoviesRecyclerView` and for the landscape view, I had `recyclerViewPopularMoviesFragment` which produces an exception.

Reply:
Now I set adapter by fragment 😄